zcash_client_backend/src/scanning.rs

```diff
@@ -505,7 +513,7 @@ pub(crate) fn scan_block_with_runner<
         if !(shielded_spends.is_empty() && shielded_outputs.is_empty()) {
             wtxs.push(WalletTx {
                 txid,
-                index: tx.index as usize,
+                index: tx_index as usize,
```

Comment:
There's an infallible [`From<u16> for usize`](https://doc.rust-lang.org/std/convert/trait.From.html#impl-From%3Cu16%3E-for-usize), so this need not use `as`.
```suggestion
                index: tx_index.into(),
```